src/Events/Extensions/CloudEventExtensions.cs

```diff
@@ -19,5 +20,26 @@ public static string Serialize(this CloudEvent cloudEvent, CloudEventFormatter f
             var bytes = formatter.EncodeStructuredModeMessage(cloudEvent, out _);
             return Encoding.UTF8.GetString(bytes.Span);
         }
+
+        /// <summary>
+        /// Retrieves the resource from the cloud event. Returns null if it isn't defined.
+        /// </summary>
+        public static string GetResource(this CloudEvent cloudEvent) => cloudEvent["resource"]?.ToString();
+
+        /// <summary>
+        /// Retrieves the resource instance from the cloud event. Returns null if it isn't defined.
+        /// </summary>
+        public static string GetResourceInstance(this CloudEvent cloudEvent) => cloudEvent["resourceinstance"]?.ToString();
+
+        /// <summary>
+        /// Sets the resource extension attribute of the cloud event
+        /// </summary>
+        public static void SetResource(this CloudEvent cloudEvent, string resource)
+        {
+            if (!string.IsNullOrEmpty(resource))
```

Comment:
This if check should ideally be reflected in the method name. `SetResourceIfNull` or something.

Reply:
We don't actually check that resource isn't defined on the cloud event, only that the input isn't empty. That being said, we probably should. Renaming and ensuring resource hasn't been defined before setting the value

Reply:
oh, sorry, my comment was a mess.

My primary concern was the hidden if. Set methods should work like a setter and set the value. Think of how you would have handled this if Resource was a regular property.

Reply:
Removed the hidden if check, and the method name reflects what it does. We don't override the resource name if it is already defined. Do you think we need an argument null exception here in addition to stop from setting an empty resource value?

Reply:
Maybe we could rename to setResourceIfNotDefined to be more precise?

Reply:
The check should be done by the call point. Which I can't find in this PR. Where is the method being used?

Reply:
Currently only used by tests. Was wondering why I implemented it at this point myself. But we will need it for #338 as well

Reply:
I'm happy with `SetResourceIfNotDefined`.